src/FieldWarning/Assets/Units/Component/Movement/MovementComponent.cs

```diff
@@ -74,14 +75,23 @@ private void Update()
             UpdateCurrentPosition();
         }
 
-        private void UpdateCurrentPosition()
+        private void UpdateCurrentPosition()  
         {
+            //update position and velocity param of this unit
+
+
+
             Vector3 diff = (_moveStrategy.NextPosition - transform.position) * Time.deltaTime;
             Vector3 newPosition = transform.position;
             newPosition.x += TRANSLATION_RATE * diff.x;
             newPosition.y = _moveStrategy.NextPosition.y;
             newPosition.z += TRANSLATION_RATE * diff.z;
 
+            Velocity.x = TRANSLATION_RATE * diff.x;
+            Velocity.y = newPosition.y - transform.position.y;
+            Velocity.z = TRANSLATION_RATE * diff.z;
+            Velocity = Velocity / Time.deltaTime;
```

Comment:
a comment on the purpose of this or of the velocity property would be very useful for future versions of you and me